pkg/operator/starter.go

```diff
@@ -119,105 +154,138 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		},
 	).WithCSIConfigObserverController(
 		"AWSEBSDriverCSIConfigObserverController",
-		configInformers,
+		guestConfigInformers,
 	).WithCSIDriverControllerService(
 		"AWSEBSDriverControllerServiceController",
 		assets.ReadFile,
 		"controller.yaml",
-		kubeClient,
-		kubeInformersForNamespaces.InformersFor(defaultNamespace),
-		configInformers,
+		controlPlaneKubeClient,
+		controlPlaneKubeInformersForNamespaces.InformersFor(controlPlaneNamespace),
+		guestConfigInformers,
 		[]factory.Informer{
-			secretInformer.Informer(),
-			nodeInformer.Informer(),
-			cloudConfigInformer.Informer(),
-			infraInformer.Informer(),
-			configMapInformer.Informer(),
+			controlPlaneSecretInformer.Informer(),
+			guestNodeInformer.Informer(),
+			controlPlaneCloudConfigInformer.Informer(),
+			guestInfraInformer.Informer(),
+			controlPlaneConfigMapInformer.Informer(),
 		},
-		csidrivercontrollerservicecontroller.WithSecretHashAnnotationHook(defaultNamespace, secretName, secretInformer),
+		withHypershiftDeploymentHook(isHypershift),
+		withHypershiftReplicasHook(isHypershift, guestNodeInformer.Lister()),
+		withNamespaceDeploymentHook(controlPlaneNamespace),
+		csidrivercontrollerservicecontroller.WithSecretHashAnnotationHook(controlPlaneNamespace, secretName, controlPlaneSecretInformer),
 		csidrivercontrollerservicecontroller.WithObservedProxyDeploymentHook(),
-		csidrivercontrollerservicecontroller.WithReplicasHook(nodeInformer.Lister()),
-		withCustomCABundle(cloudConfigLister),
-		withCustomTags(infraInformer.Lister()),
-		withCustomEndPoint(infraInformer.Lister()),
+		withCustomAWSCABundle(guestInfraInformer.Lister(), controlPlaneCloudConfigLister),
```

Comment:
Since other hooks pass `isHypershift` directly, would it make sense to pass it here and not read it from `guestInfraInformer`?

Reply:
Makes sense, this predates the `isHypershift` trick. Fixed.